### PR TITLE
Integration Tests (1st batch)

### PR DESCRIPTION
**Pull request by jtourkos**
- Refactored `NFTDriverClient` & `DripsHubClient` creation API. 
- Added `NFTDriverClient` integration tests.

### CODE REVIEW

File: src/NFTDriver/NFTDriverClient.ts [hunk 1]
```diff
@@ -56,40 +57,48 @@ export default class NFTDriverClient {
 	// TODO: Update the supported chains documentation comments.
 	/**
 	 * Creates a new immutable `NFTDriverClient` instance.
-	 * @param  {JsonRpcProvider} provider The network provider.
+	 * @param  {JsonRpcProvider} provider The network provider. It cannot be changed after creation.
```
Comment (efstajas) [inline src/NFTDriver/NFTDriverClient.ts:60]: This should probably be `{Provider}`, right? (Could also just remove the jsdoc typings and they'd be inferred from the TS function signature)

File: src/NFTDriver/NFTDriverClient.ts [hunk 2]
```diff
@@ -56,40 +57,48 @@ export default class NFTDriverClient {
...
 	 *
-	 * The `provider` must have a `signer` associated with it, and cannot be changed after creation.
-	 *
-	 * The supported networks are:
+	 * The `provider` must be connected to one of the following supported networks:
 	 * - 'goerli': chain ID `5`
 	 * - 'polygon-mumbai': chain ID `80001`
+	 * @param  {JsonRpcProvider} signer The singer used to sign transactions. It cannot be changed after creation.
```
Comment (efstajas) [inline src/NFTDriver/NFTDriverClient.ts:65]: should be typed as signer?

File: src/DripsHub/DripsHubClient.ts [hunk 3]
```diff
@@ -54,73 +54,50 @@ export default class DripsHubClient {
...
-	 * @throws {@link DripsErrors.argumentError} if the `provider.signer` is missing.
-	 * @throws {@link DripsErrors.unsupportedNetworkError} if the `provider` is connected to an unsupported network.
+	 * @returns A `Promise` which resolves to the new client instance.
+	 * @throws {@link DripsErrors.clientInitializationError} if the client initialization fails.
 	 */
 	public static async create(
-		signer: JsonRpcSigner,
+		provider: Provider,
+		signer: Signer | undefined = undefined,
 		customDriverAddress: string | undefined = undefined
```
Comment (efstajas) [inline src/DripsHub/DripsHubClient.ts:73]:
```suggestion
		signer?: Signer,
		customDriverAddress?: string,
```

**Review by efstajas — CHANGES_REQUESTED** [review]

File: src/NFTDriver/NFTDriverClient.ts (hunk 2 above)
Comment (jtourkos) [reply src/NFTDriver/NFTDriverClient.ts:65]: Yes, changed.

File: src/NFTDriver/NFTDriverClient.ts [hunk 4]
```diff
@@ -56,40 +57,48 @@ export default class NFTDriverClient {
...
-	 * @throws {@link DripsErrors.addressError} if the `provider.signer`'s address is not valid.
-	 * @throws {@link DripsErrors.argumentError} if the `provider.signer` is missing.
-	 * @throws {@link DripsErrors.unsupportedNetworkError} if the `provider` is connected to an unsupported network.
+	 * @throws {@link DripsErrors.clientInitializationError} if the client initialization fails.
 	 */
 	public static async create(
-		provider: JsonRpcProvider,
+		provider: Provider,
+		signer: Signer,
 		customDriverAddress: string | undefined = undefined
```
[the remark this answers is not in the document]
Comment (jtourkos) [reply src/NFTDriver/NFTDriverClient.ts:76]: ✅

**Comment by jtourkos** [comment]
I applied the changes and also changed the waiting time to use [this](https://github.com/radicle-dev/drips-app-v2/blob/main/src/lib/utils/expect.ts). Nice tip @efstajas!

**Pull request merged**